packages/schematics/angular/guard/index.ts

```diff
@@ -37,13 +37,26 @@ export default function (options: GuardOptions): Rule {
       options.path = buildDefaultPath(project);
     }
 
+    let implementations = '';
+    let implementationImports = '';
+    if (options.implements && options.implements.length > 0) {
```

Comment:
set `options.implements` to an empty array if undefined first to avoid the repeat checks